src/components/dragZoomLayer.ts

```diff
@@ -0,0 +1,152 @@
+module Plottable {
+export module Components {
+  export class DragZoomLayer extends Components.SelectionBoxLayer {
+    private _dragInteraction: Interactions.Drag;
+    private _doubleClickInteraction: Interactions.DoubleClick;
+    private xDomainToRestore: Numeric[];
+    private yDomainToRestore: Numeric[];
+    private isZoomed = false;
+    private easeFn: (t: number) => number = d3.ease("cubic-in-out");
+    private _animationTime = 750;
+
+    /* Constructs a SelectionBoxLayer with an attached DragInteraction and DoubleClickInteraction.
+     * On drag, it triggers an animated zoom into the box that was dragged.
+     * On double click, it zooms back out to the original view, before any zooming.
+     * The zoom animation uses an easing function (default d3.ease("cubic-in-out")) and is customizable.
+     * Usage: Construct the selection box layer and attach x and y scales, and then add the layer
+     * over the plot you are zooming on using a Component Group.
+     */
+    constructor(xScale: QuantitativeScale<Numeric>,
```

Comment:
now that Category scales support pan/zoom maybe we could instead pass a panzoominteraction here? Or pass in TransformableScale types?